src/Handler/Home.hs

```diff
@@ -1,21 +1,27 @@
 module Handler.Home where
 
-import Import
 import Yesod.Paginator
-import Yesod.Paginator.Simple
+    ( PageWidget
+    , PageWidgetConfig(..)
+    , defaultPageWidgetConfig
+    , simplePaginationWidget
+    )
+
+import Import
+import Query
 
 import Handler.ScreamDetail (singleScream)
 
 getHomeR :: Handler Html
 getHomeR = do
-    (screams, pagination) <- fetchScreams
+    (screams', pagination) <- runDB $ paginatedScreams 30 simpleWidget
+    images <- runDB $ fetchImagesForScreams screams'
+    let screams = joinOneToMany screamImage screams' images
```

Comment:
This feels so close to a nicer abstraction ... maybe Pat or Joe can see something I can't.